python/ql/test/experimental/dataflow/tainttracking/basic/test.py

```diff
@@ -6,3 +6,21 @@
 def func():
     also_tainted = SOURCE
     SINK(also_tainted)
+
+
+# Various instances where flow is undesirable
+
+tainted2 = NON_SOURCE
+SINK(tainted2)
+
+def write_global():
+    global tainted2
+    tainted2 = SOURCE
+
+tainted3 = SOURCE
+tainted3 = NON_SOURCE
+SINK(tainted3)
+
+def use_of_tainted3():
+    global tainted3
+    tainted3 = NON_SOURCE
```

Comment:
Very good to see these tests, but I think the output is a bit verbose, and you have to do some real detective work to see what the results are :detective:

Can we move these out of the `/basic/` folder and adopt the `ensure_not_tainted` as sinks? :pray:

Reply:
Hm. `TestTaintLib` expects tests to appear inside functions, but that doesn't fit this use case (since the behaviour at the top level is very different from inside a function). I'll see if I can reconfigure things in `TestTaintLib` so this works.

Reply:
yeah, that's on me :grimacing: but I'm very happy to see the fix was very easy :stuck_out_tongue: